### PR TITLE
Partial #19687: Create download artifacts script and other CI changes

### PR DESCRIPTION
**Pull request by prodrigestivill**
This pull request responses to #19687 and improves CI changes performed in #19695.

Features:
1. Adds a new script in `sys/download-artifacts.sh` to download all artifacts into `dist/artifacts` using the [gh cli tool ](https://github.com/cli/cli#readme).
2. Renames all CI artifacts names to match the job name that publishes it.
3. Move jobs from CI workflow to external workflows (like it was with freebsd and windows)
4. Add 30 days retention policy to all generated artifacts (not to the release assets)

Any comment or feedback will be welcome if any folder or change don't match with the philosophy of the project.

### CODE REVIEW

File: sys/download-artifacts.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+set -e
+
+WORKFLOWS="linux apple android windows freebsd"
+DESTDIR="dist/artifacts"
+LIMIT=100
+
+cd `dirname $PWD/$0`/..
+
+COMMIT="$1" # Optional
+if [ -z "${COMMIT}"]; then
```
Comment (radare) [inline sys/download-artifacts.sh:12]:
```suggestion
if [ -z "${COMMIT}" ]; then
```

File: sys/download-artifacts.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+
+set -e
+
+WORKFLOWS="linux apple android windows freebsd"
+DESTDIR="dist/artifacts"
+LIMIT=100
+
+cd `dirname $PWD/$0`/..
+
```
Comment (radare) [inline sys/download-artifacts.sh:10]: Can you check if `gh` exists and works before going further into the loop

**Review by radare — APPROVED** [review]

**Comment by prodrigestivill** [comment]
Updated "Move download artifacts as script" commit with the requested changes.

**Comment by trufae** [comment]
Beautiful! <3 Thanks for the improvements!

**Pull request merged**